include/libtorrent/kademlia/get_item.hpp

```diff
@@ -43,7 +43,7 @@ namespace libtorrent { namespace dht
 class get_item : public find_data
 {
 public:
-	typedef boost::function<bool(item&, bool)> data_callback;
+	typedef boost::function<void(item&, bool)> data_callback;
```

Comment:
right. it makes sense that you don't trigger the put until you get an authoritative response. Traditionally though, the nodes_callback is called when we receive a response without any data (or am I misremembering that?). i.e. it just contains "nodes" to continue the traversion, but no "values" with bittorrent peers. I'm trying to understand why it makes sense to call this callback nodes_callback. It seems like it should be something like completed_callback (or something along those lines).